csrc/quantization/quantizer.cu

```diff
@@ -103,28 +96,31 @@ __global__ void quantize_kernel(float* vals, int group_size, int num_bits)
     int warp_num = blockDim.x >> 5;
     int id = threadIdx.x;
 
-    float4* vals_cast = reinterpret_cast<float4*>(vals);
+    constexpr int granularity = 16;
+    constexpr int vals_per_access = granularity / sizeof(float);
 
-    float4 data[MAX_REG];
+    float data[MAX_REG * vals_per_access];
```

Comment:
Where applicable, apply the comments for the `__half` implementation to the `float` implementation.